File: my-solutions/0283-move-zeroes/solution.scala
object Solution {
    def moveZeroes(nums: Array[Int]): Unit = {
        var nonZeroPointer:Int = 0

        for(el <- 0 until nums.length){
            if(nums(el) != 0) {
                nums(nonZeroPointer) = nums(el)
                nonZeroPointer += 1
            }
        }
        
        for(el <- nonZeroPointer until nums.length){
            nums.update(el, 0)
        }
    }
}

File: my-solutions/0283-move-zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        non_zero_ptr = 0

        for i in range(len(nums)):

            if nums[i] != 0:
                nums[non_zero_ptr], nums[i] = nums[i], nums[non_zero_ptr]
                non_zero_ptr += 1

        return nums
